Templates/Common-Backend-Scripts/utilities/fetchBuildDependencies.groovy

```diff
@@ -0,0 +1,421 @@
+if (applicationDescriptorUtilsFile.exists()) {
+	applicationDescriptorUtils = loadScript(applicationDescriptorUtilsFile)
+} else {
+	println("*! [ERROR] The Application Descriptor Helper script '$applicationDescriptorUtilsFile.getName()' was not found. Exiting.")
+	System.exit(8)
+}
+
+File applicationDescriptorFile = new File(props.applicationDescriptor)
+if (!applicationDescriptorFile.exists()) {
+	println("*! [ERROR] The Application Descriptor file '${applicationDescriptorFile.getName()}' doesn't exist. Exiting.")
+	System.exit(8)
+}
+
+// setup import directory
+File importFolder = new File("$props.workspace/imports")
+File baselineFolder = new File("$props.workspace/baseline")
+if (importFolder.exists()) importFolder.deleteDir()
+importFolder.mkdirs()
+
+// setup package cache
+tmpPackageDir = (props.enablePackageCache && props.enablePackageCache.toBoolean() && props.packageCacheLocation) ? new File(props.packageCacheLocation) : new File("$props.workspace/imports_download/")
+if (!tmpPackageDir.exists()) tmpPackageDir.mkdirs()
+
+// Parse the application descriptor and application configurations based on the defined schema
+applicationDescriptor = applicationDescriptorUtils.readApplicationDescriptor(applicationDescriptorFile)
+
+
+ArrayList<ExternalDependency> externalDependencies = new ArrayList<>()
+
+if (applicationDescriptor.dependencies) {
+
+	println("*** Fetching external application packages")
+
+	// Loop through all dependencies found in AD
+	applicationDescriptor.dependencies.each { dependency ->
+
+		// validate dependency record
+		assert dependency.type : "Missing dependency type attribute in dependency record"
+		assert dependency.reference : "Missing dependency reference attribute in dependency record"
+		assert dependency.buildid : "Missing buildid attribute in dependency record"
+		assert dependency.name : "Missing name attribute in dependency record"
+
+		// compute tar file name based on build type
+		if (dependency.type.equalsIgnoreCase("release")) {
+			props.put("tarFileName","${dependency.name}-${dependency.reference}-${dependency.buildid}.tar")
+		} else {
+			props.put("tarFileName","${dependency.name}-${dependency.buildid}.tar")
+		}
+
+		// Construct the path within the Artifact repo
+		repositoryName="${dependency.name}-${props.artifactRepositoryNameSuffix}"
+
+		// Prepare for url computation using method in ArtifactRepoHelpers
+		props.put("versionName","${dependency.reference}") // compute the version name being part of the path
+		props.put("artifactRepository.directory", "${dependency.type}") // compute the main directory to classify builds
+		props.put("artifactRepository.repo", "${repositoryName}") // Artifact repository name
+
+		// The absolute url the package in artifact repo
+		artifactUrl = artifactRepositoryHelpers.computePackageUrl(props)
+
+		// retrieve path without artifact url
+		artifactRelPath = artifactUrl.replaceAll(props.get("artifactRepository.url"),"")
```

Comment:
Why not having a method `computeRelativePackageUrl()` in artifactRepositoryHelpers.groovy, instead of playing with replacing substrings?